common/beerocks/bcl/source/network/network_utils.cpp

```diff
@@ -767,6 +770,54 @@ bool network_utils::linux_iface_is_up_and_running(const std::string &iface)
     return (false);
 }
 
+bool network_utils::linux_iface_get_speed(const std::string &iface, uint32_t &speed)
+{
+    int sock;
+    bool result = false;
+
+    speed = SPEED_UNKNOWN;
+
+    sock = socket(PF_INET, SOCK_DGRAM, IPPROTO_IP);
+    if (sock < 0) {
+        LOG(ERROR) << "Can't create SOCK_DGRAM socket: " << strerror(errno);
+    } else {
+        struct ifreq ifr;
+        struct ethtool_link_settings settings;
+        int rc;
+
+        beerocks::string_utils::copy_string(ifr.ifr_name, iface.c_str(), sizeof(ifr.ifr_name));
+        ifr.ifr_data = (char *)&settings;
```

Comment:
I think we need `reinterpret_cast<char*>` here?
C++ people tend not to like C-style casts ;-)

Reply:
Agree with you but looks like this is common practice here and C-style casts are spread all over the project. Maybe @arnout or @tomereli can give their opinion/version on this and decide if I should change it or stick to C-style

Reply:
If I may add my view, I don't think there is any good reason to use C-style casts anywhere in the project. The ones that are already there probably either were from the beginning, or slipped through during reviews.
There are also a lot of `reinterpret_cast` used all over the project, and almost/all of them could be `static_cast`s.

I think this is another good reason to add static analysis tools to at the very least catch C-style casts in C++ code, and if possible warn for `reinterpret_cast`s.

Reply:
C style casts should be avoided. The ones that are there should be refactored to use c++ casts in the future.

Reply:
Done replacing C-style class with reinterpret_cast